mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -425,7 +426,9 @@ def create_model_version(
             notebook_entity = Notebook(id=str(notebook_id))
             entity = Entity(notebook=notebook_entity)
             lineage_header_info = LineageHeaderInfo(entities=[entity])
-            extra_headers = {_DATABRICKS_LINEAGE_ID_HEADER: message_to_json(lineage_header_info)}
+            header_dict = json.loads(message_to_json(lineage_header_info))
+            header_json = json.dumps(header_dict)
```

Comment:
Can we insert an inline comment explaining what we're doing here? The fact that this removes newlines might not be immediately obvious to folks who refactor this code later and wonder why we parse JSON as a dict just to dump it back.

In the comment, can we also explain why the JSON header content can't contain newlines?